src/EventSubscriber/StrawberryfieldEventPresaveSubscriberAsFileStructureGenerator.php

```diff
@@ -277,10 +259,20 @@ public function onEntityPresave(StrawberryfieldCrudEvent $event) {
    *     "ismemberof"
    *   }
    *
+   *  @param array $for_from_as
+   *    An array with all source keys (dr:from) fetched from the flat array
+   *    This may not exist of course.
+   *
    * @return array
    *   The cleaned/up $entityMapping
    */
-  private function cleanUpEntityMappingStructure(array $fullvalues) {
+  private function cleanUpEntityMappingStructure(array $fullvalues, array $for_from_as) {
+
+    // If not present of empty and we do have dr:for try to rebuild from that
+    // Its an edge cases but adds an extra level of dexterity/safety.
+    if (!isset($fullvalues['ap:entitymapping']['entity:file']) && !empty($for_from_as)) {
```

Comment:
If `$for_from_as` can be empty, I think it should be initialized to [] in the signature.

Reply:
Thanks. The caller does the checks here but for sanity I gave it a default